tutorials/nginx-ingress-gke/index.md

```diff
@@ -0,0 +1,350 @@
+
+This guide explains how to deploy the [Nginx Ingress
+Controller](https://github.com/kubernetes/ingress-nginx) on Google Kubernetes
+Engine. 
+
+In Kubernetes,
+[Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+allows external users and client applications access to HTTP services.  Ingress
+consists of two components. 
+[Ingress Resource](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource)
+is a collection of rules for the inbound traffic to reach Services.  These are
+Layer 7 (L7) rules that allow hostnames (and optionally paths) to be directed to
+specific Services in Kubernetes.  The second component is the
+[Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers)
+which acts upon the rules set by the Ingress Resource, typically via an HTTP or
+L7 load balancer.  It is vital that both pieces are properly configured to route
+traffic from an  outside client to a Kubernetes Service.  
+
+Nginx is a popular choice for an Ingress Controller for a variety of features:
+
+-  [Websocket](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/websocket),
+   which allows you to load balance Websocket applications.
+-  [SSL Services](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/ssl-services),
+   which allows you to load balance HTTPS applications.
+-  [Rewrites](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/rewrites),
+   which allows you to rewrite the URI of a request before sending it to the
+   application.
+-  [Session Persistence](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/session-persistence)
+   (Nginx Plus only), which guarantees that all the requests from the same
+   client are always passed to the same backend container.
+-  [Support for JWTs](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/jwt)
+   (Nginx Plus only), which allows Nginx Plus to authenticate requests by
+   validating JSON Web Tokens (JWTs).
+
+The following diagram shows the architecture described above:
+
+<img src="https://github.com/ameer00/community/blob/master/tutorials/nginx-ingress-gke/Nginx%20Ingress%20on%20GCP%20-%20Fig%2002.png" width="65%">
+
+This tutorial illustrates how to set up a
+[Deployment in Kubernetes](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)
+with an Ingress Resource using Nginx as the Ingress Controller to route/load
+balance traffic from external clients to the Deployment.  This tutorial explains how to accomplish the following:
+
+-  Create a Kubernetes Deployment
+-  Set up an Ingress Resource object for the Deployment
+-  Set up an Ingress Controller using Nginx as the controller (instead of the
+   Kubernetes Engine default Cloud Load Balancer)
+
+## Objectives
+
+-  Create a simple Kubernetes web _application_ Deployment 
+-  Create a _default backend_ Deployment to serve 404 for all non-matching
```

Comment:
Since there was an nginx default backend deployment, I wanted to highlight it for educational purposes, but I agree with you on the premise that it is not part of Ingress.  Most of my customers have deployed it this way so I figured I would add it

Reply:
According to this https://github.com/kubernetes/ingress-nginx 
Under the Requirements section, it mentions a default backend service.
I have also read multiple articles that all deploy this service.
Are you asking me to completely remove the default backend service?

Reply:
No I'm just saying this is not a target objective, it's just a detail and I'm not sure if it deserves its own section.

If you just use the Helm chart, there wouldn't be a need for this section in the first place.

Reply:
I totally get your point.  However, I think its a matter of opinion and a different way of doing it.  I learn and deploy from doing things in a most basic way with the most basic tools that everyone has access to (kubectl e.g.) and then think of streamlining the process (using helm for example).  Not everyone has or uses helm.  However, I do think its worth mentioning that helm charts are maintained and supported by a community and thats an option to use but I could argue its not the only way.  I think what we are talking about here is subjective and a matter of opinion which may differ from person to person :)

Reply:
That's a fair point. I don't want to enforce you to change the content. However, I can give you some perspective on what happens if this is an example that keeps aging in your personal repo:

Since the nginx controller lives in your repo and it's likely not up-to-date all the time and doesn't have the latest tweaks in terms of production-readiness, I personally will not be sending people to use this solution and will likely develop my own version which uses Helm and relies on Helm community's maintenance of this chart.

Reply:
I have changed it to use stable helm chart as per your suggestion